### ISSUE TITLE
Detecting extra `overload` entries after `*`

### ISSUE THREAD

**Issue by jamesbraza** [python/mypy#20278]
**Feature**

`mypy==1.18.2` is not detecting the `b` as extra here after the `*`:

```python
from typing import overload


@overload
def some_method(a: int, **kwargs) -> None: ...
@overload
def some_method(a: str, *, b: None = None, **kwargs) -> None: ...


def some_method(a: int | str, **kwargs) -> None:
    pass
```

It may also be that due to `**kwargs`, that a type checker can't do anything here.

**Pitch**

Detecting extra `overload` parameters to keep code clean.

**Comment by A5rocks** [comment]
Sorry, I don't quite follow. What would it mean for mypy to detect something as extra? (I don't know the terminology you're using...)

**Comment by jamesbraza** [comment]
Note that the second `overload` has an extra `b: None = None`. Perhaps it can be detected as extra and removed

**Comment by A5rocks** [comment]
Sorry I still don't follow. What makes `b` extra? If you remove it, `some_method("5", b=4)` becomes valid?

**Comment by jamesbraza** [comment]
The reason it's extra is `b` is not in the core signature of `some_method` at the bottom, it's just accidentally present in an `overload`

**Comment by A5rocks** [comment]
But this is intentional! For instance you can do this:
```py
@overload
def f(x: int) -> str: ...

@overload
def f(y: str) -> int: ...

def f(*args, **kwargs) -> object: ...
```

And if your overloads use a parameter not allowed by the actual signature, mypy already warns.

**Comment by sterliakov** [comment]
Yep, the only restriction is that every overload signature must be compatible with the implementation signature. If any call of the overload signature is a valid call of the impl signature (as in your case), the def is correct. This is allowed on purpose - to provide additional restrictions on allowed argument combinations if needed.
